source/OptimaJet.DWKit.StarterApplication/Services/BuildEngine/BuildEngineBuildService.cs

```diff
@@ -9,29 +9,37 @@
 using System.Threading.Tasks;
 using System.Collections.Generic;
 using OptimaJet.DWKit.StarterApplication.Utility;
+using Newtonsoft.Json;
+using System.Net;
 
 namespace OptimaJet.DWKit.StarterApplication.Services.BuildEngine
 {
     public class BuildEngineBuildService: BuildEngineServiceBase
     {
         public IRecurringJobManager RecurringJobManager { get; }
         public WebRequestWrapper WebRequestWrapper { get; }
+        public IWebClient WebClient { get; }
         public IJobRepository<Product, Guid> ProductRepository { get; }
         public IJobRepository<ProductArtifact> ProductArtifactRepository { get; }
+        public IJobRepository<ProductBuild> ProductBuildRepository { get; }
 
         public BuildEngineBuildService(
             IRecurringJobManager recurringJobManager,
             IBuildEngineApi buildEngineApi,
             WebRequestWrapper webRequestWrapper,
+            IWebClient webClient,
             IJobRepository<Product, Guid> productRepository,
             IJobRepository<ProductArtifact> productArtifactRepository,
+            IJobRepository<ProductBuild> productBuildRepository,
             IJobRepository<SystemStatus> systemStatusRepository
         ) : base(buildEngineApi, systemStatusRepository)
         {
             RecurringJobManager = recurringJobManager;
             WebRequestWrapper = webRequestWrapper;
```

Comment:
Eventually, I would like to see WebRequestWrapper be more generic (like IWebClient) instead of being specific to ProductArtifact.